homeassistant/components/switch/rest.py

```diff
@@ -75,32 +81,45 @@ def is_on(self):
 
     def turn_on(self, **kwargs):
         """Turn the device on."""
+        body_on_t = template.render(self._hass, self._body_on)
         request = requests.post(self._resource,
-                                data=self._body_on,
+                                data=body_on_t,
                                 timeout=10)
         if request.status_code == 200:
             self._state = True
         else:
             _LOGGER.error("Can't turn on %s. Is resource/endpoint offline?",
-                          self._resource)
+                          -                          self._resource)
 
     def turn_off(self, **kwargs):
         """Turn the device off."""
+        body_off_t = template.render(self._hass, self._body_off)
         request = requests.post(self._resource,
-                                data=self._body_off,
+                                data=body_off_t,
                                 timeout=10)
         if request.status_code == 200:
             self._state = False
         else:
             _LOGGER.error("Can't turn off %s. Is resource/endpoint offline?",
-                          self._resource)
+                          -                          self._resource)
 
     def update(self):
         """Get the latest data from REST API and update the state."""
         request = requests.get(self._resource, timeout=10)
-        if request.text == self._body_on:
-            self._state = True
-        elif request.text == self._body_off:
-            self._state = False
+
+        if self._value_template is not None:
+            response = template.render_with_possible_json_value(
+                self._hass, self._value_template, request.text, False)
```

Comment:
Right now you're returning `False` if an error occurs. Which then will not match any of your if-statements (they compare to a string) and `self._state` is set to `None`. Is this expected?